File: cir-villalobos/e32/Mamifero.cpp
#include "Mamifero.h"

Mamifero::Mamifero(){};
Mamifero::Mamifero(int nPatas, std::string col,
              std::string nom, std::string esp)
         : numeroDePatas(nPatas), color(col), nombre(nom), especie(esp){}

Mamifero::~Mamifero(){}

bool Mamifero::setPatas(int nPatas){
    if(!(nPatas >= 0))    return false;

    this->numeroDePatas = nPatas;
    return true;
}

bool Mamifero::setColor(std::string col){
    if(col.length() == 0)    return false;
    
    this->color = col;
    return true;
}

bool Mamifero::setNombre(std::string nom){
    if(nom.length() == 0)    return false;
    
    this->nombre = nom;
    return true;
}

bool Mamifero::setEspecie(std::string esp){
    if(esp.length() == 0)    return false;
    
    this->especie = esp;
    return true;
}


int Mamifero::getPatas()   {    return this->numeroDePatas;}
auto Mamifero::getColor()  {    return this->color;        }
auto Mamifero::getNombre() {    return this->nombre;       }
auto Mamifero::getEspecie(){    return this-> especie;     }

File: 20A40/cir-Israel/Ejercicio 32/Mamifero.java
public class Mamifero {
    private int patas;
    private String color;
    private String especie;
    
    public Mamifero() {}
    
    public Mamifero(String e, int p, String c) {
      especie = e;
      patas = p;
      color = c;
    }
    
    public boolean setEspecie(String e) {
      if (e.length() != 0) {
        especie = e;
        return true;
      } else return false;
    }
    
    public boolean setPatas(int p) {
      if (p >= 0 && p <= 4) {
        patas = p;
        return true;
      } else return false;    
    }
    
    public boolean setColor (String c) {
      if (c.length() != 0) {
        color = c;
        return true;
      } else return false;
    }
    
    public int getPatas() {  return patas;  }
    public String getColor() {  return color;  }
    public String getEspecie() {  return especie;  }
}